07-asymmauth.tex

```diff
@@ -191,8 +191,8 @@ \subsection{Schlüssel- und Signaturerzeugung}
 Zahl $e \in \{1, \dots, p - 1\}$ gewählt,
 wobei $p=|\G|$. Damit berechnet man 
 \begin{align*}
-a &:= g^e \in \mathbb{G}\\
-b &:= (M - a \cdot x) \cdot e^{-1} \mod p
+a &:= g^e \mod p \in \mathbb{G}\\
```

Comment:
Versteh die Frage nicht ganz, glaube ich. Falls durch meine Antwort nicht alles klar wird, frag bitte nochmal nach!

Die Gruppenordnung ist p (das muss nicht unbedingt eine Primzahl sein), deswegen kann man die Exponenten als aus Z_p betrachten. p ist hier nicht doppeldeutig (zumindest an dieser Stelle, leider wird es aber an anderer Stelle undeutlich verwendet, wie mir gerade aufgefallen ist). Das folgt aus verschiedenen Sätzen aus der Gruppentheorie.
Im Prinzip ist G zyklisch und hat p verschiedene Elemente g^1, g^2, g^3, ... g^p = g^0 - deswegen rechnet man im Exponenten quasi modulo p.
Unten muss nicht modulo der Gruppenordnung gerechnet werden, da a ein Gruppenelement ist (im Gegensatz zu b - b ist "nur" ein Exponent, weshalb hier modulo der Gruppenordnung gerechnet wird). Nicht alle Gruppen haben als Gruppenoperation die Addition oder Multiplikation modulo irgendwas. Die Gruppe hier ist schon "Z Modulo Irgendwas" (in einer anderen Gruppe machen EL-Gamal-Signaturen nicht wirklich Sinn, dazu gleich mehr), deswegen könnte man das schon hinschreiben, wir haben uns für die abstrakt-formale Definition aber dagegen entschieden, damit klarer ist, wann wir "im Exponent" und wann "in der Gruppe" rechnen.

Hier kann verwirrend sein, dass a einmal als "Zahl" modulo Z_N interpretiert wird, um b zu berechnen, obwohl a eigentlich formal ein Gruppenelement ist. Das geht hier nur, weil wir als Gruppe schon Z_N verwenden (sonst macht das alles relativ wenig Sinn, weil dann unklar ist, wie man ein Gruppenelement effizient als Zahl interpretiert). Das a so doppeldeutig (einmal als "Zahl", einmal als Gruppenelement) verwendet wird, ist eine merkwürdige Eigenart des El-Gamal-Signaturverfahrens und aus formal-mathematischer Sicht auch etwas fragwürdig.

Viele Grüße,
Björn

Am 23.07.2016 um 22:23 schrieb Mark Giraud:

In 07-asymmauth.tex<https://github.com/skript-sicherheit/skript/pull/23#discussion_r71977741>:

> @@ -191,8 +191,8 @@ \subsection{Schlüssel- und Signaturerzeugung}
>  Zahl $e \in \{1, \dots, p - 1\}$ gewählt,
>  wobei $p=|\G|$. Damit berechnet man
>  \begin{align*}
> -a &:= g^e \in \mathbb{G}\\
> -b &:= (M - a \cdot x) \cdot e^{-1} \mod p
> +a &:= g^e \mod p \in \mathbb{G}\\


p=|G| ist hier die Ordnung der Gruppe. Allerdings müsste man hier modulo die Primzahl rechnen, oder nicht?
Bzw p ist hier doppeldeutig, da einmal in Z_p verwendet und dann als Gruppenordnung redefiniert.
Und unten dann modulo gruppenordnung also mod p

-
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://github.com/skript-sicherheit/skript/pull/23/files/cc42ae29778a400d6e762539c979c23046587cbd..a1640dad032432a08a735fa6db7ba734eb7f8f8a#r71977741>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AH_qn99IXRjfA4SAvHLJ1S9lDNEKNOs0ks5qYnhOgaJpZM4JSgFR>.